lib/quantizeAttributes.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+
+var addExtensionsUsed = require('./addExtensionsUsed');
+var byteLengthForComponentType = require('./byteLengthForComponentType');
+var getAccessorByteStride = require('./getAccessorByteStride');
+var numberOfComponentsForType = require('./numberOfComponentsForType');
+var packBuffers = require('./packBuffers');
+
+module.exports = quantizeAttributes;
+
+/**
+ * Quantizes attributes in the provided glTF using the WEB3D_quantized_attributes extension.
+ * If options is undefined, all quantizable attributes will be quantized with the default options.
+ * If options.attributes is undefined, all quantizable attributes will be quantized with the provided options.
+ * If options.precision is undefined, the decodeMatrix be written in its full form.
+ *
+ * @param {Object} [gltf] A javascript object holding a glTF hierarchy.
+ * @param {Object} [options=undefined] Defines more specific quantization behavior.
+ * @param {Object} [options.semantics=undefined] Defines which semantics should be quantized.
+ * @param {Number} [options.precision=undefined] Restricts the number of decimal places in the decodeMatrix.
+ *
+ * @returns glTF with quantized attributes
+ *
+ * @example
+ * var gltf = {
+ *     accessors: {
+ *         accessor_0: {...}
+ *         accessor_1: {...}
+ *     },
+ *     meshes: {
```

Comment:
Did you see https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/56#discussion_r66105451

Perhaps the right way to keep the interface simple and pragmatic and also allow testing the corner cases is to allow the test to pass in an undocumented callback function that takes the right parameters to decide if an attribute should be quantized or not.  That keeps this implementation simple and moves the corner-case complexity (and overall less complexity) to the tests where it is the focus.